overlord/devicestate/handlers_gadget.go

```diff
@@ -212,3 +213,111 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 
 	return nil
 }
+
+func (m *DeviceManager) updateGadgetCommandLine(t *state.Task, st *state.State, isUndo bool) (updated bool, err error) {
+	snapsup, err := snapstate.TaskSnapSetup(t)
+	if err != nil {
+		return false, err
+	}
+	devCtx, err := DeviceCtx(st, t, nil)
+	if err != nil {
+		return false, err
+	}
+	if devCtx.Model().Grade() == asserts.ModelGradeUnset {
+		// pre UC20 system, do nothing
+		return false, nil
+	}
+	// but when undoing, we use the current gadget which should have
+	// been restored
+	currentGadgetData, err := currentGadgetInfo(st, devCtx)
+	if err != nil {
+		return false, err
+	}
+	var gadgetData *gadget.GadgetData
+	if !isUndo {
+		// when updating, command line comes from the new gadget
+		gadgetData, err = pendingGadgetInfo(snapsup, devCtx)
+	} else {
+		// when undoing command line comes from the current gadget
+		gadgetData = currentGadgetData
+	}
+	updated, err = boot.UpdateCommandLineForGadgetComponent(devCtx, gadgetData.RootDir)
+	if err != nil {
+		return false, fmt.Errorf("cannot update kernel command line from gadget: %v", err)
+	}
+	return updated, nil
+}
+
+func (m *DeviceManager) doUpdateGadgetCommandLine(t *state.Task, _ *tomb.Tomb) error {
+	if release.OnClassic {
+		return fmt.Errorf("cannot run update gadget kernel command line task on a classic system")
+	}
+
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var seeded bool
+	err := st.Get("seeded", &seeded)
+	if err != nil && err != state.ErrNoState {
+		fmt.Printf("not needed\n")
+		return err
+	}
+	if !seeded {
+		// do nothing during first boot & seeding
+		return nil
+	}
+
+	const isUndo = false
+	updated, err := m.updateGadgetCommandLine(t, st, isUndo)
+	if err != nil {
+		return err
+	}
+	if !updated {
+		t.Logf("no kernel command line update from gadget")
```

Comment:
```suggestion
		t.Debugf("no kernel command line update from gadget")
```

Reply:
+1

Reply:
I've updated the code here and at the place of the other comment to use logger.Debugf()